File: ruby/anagram/anagram.rb
class Anagram
  attr_reader :anagrams

  def initialize(word)
    @word = word.downcase
  end

  def match(options)
    @anagrams = options.select do |option|
      option.downcase.chars != @word.chars && option.downcase.chars.sort == @word.chars.sort
    end
  end

end

File: python/anagram/anagram.py
def detect_anagrams(word, options):
    return [option for option in options if is_anagram(word, option)]

def alphebatized(word):
    return ''.join(sorted(word))

def is_anagram(word, option):
    if word.lower() == option.lower():
        return False
    elif alphebatized(word.lower()) == alphebatized(option.lower()):
        return True
    return False
